js/map.js

```diff
@@ -134,21 +138,65 @@
       photosElementContainer.appendChild(newImgElement);
     });
 
-
     return newCardElement;
   }
 
   function renderCard(data, elementContainer) {
     elementContainer.insertBefore(data, elementContainer.firstElementChild);
   }
 
-  mapElement.classList.remove('map--faded');
+  function mouseUpHendler() {
+    mapElement.classList.remove('map--faded');
+    adFormContainer.classList.remove('ad-form--disabled');
+    updateAdress(getPinLocation(mapElement, '.map__pin--main', PIN_HIGHT, HALF_PIN_WIDTH), formAddressInput);
+    renderPins(generatePins(pinElementTemplate, nearbyLisitings), mapPinsElement);
+  }
+
+  function updateAdress(position, element) {
+    element.value = position.top + ', ' + position.left;
+  }
+
+  function getPinLocation(container, pinName, pinHight, halfPinWidth) {
+    var pinContainerElement = container.querySelector(pinName);
+    return {
+      top: (parseFloat(getComputedStyle(pinContainerElement).top) + parseFloat(getComputedStyle(pinContainerElement, 'after').top) + pinHight),
+      left: (parseFloat(getComputedStyle(pinContainerElement).left) + parseFloat(getComputedStyle(pinContainerElement, 'after').left) + halfPinWidth)
+    };
+  }
+
+  function showCard(elem, arr) {
+    renderCard(generateCard(arr[+elem.dataset.index], cardElementTemplate), filterElementContainer);
+  }
+
+  function removeCard() {
+    filterElementContainer.querySelector('.map__card').remove();
+  }
+
+  function pinsHandler(event) {
+    var button = event.target.parentElement;
+    if (button.tagName === 'BUTTON' && button.hasAttribute('data-index')) {
+      if (filterElementContainer.querySelector('.map__card')) {
+        removeCard();
+      }
+      showCard(button, nearbyLisitings);
+    }
+    return;
+  }
+
+  function closeCardHandler(event) {
+    if ((event.target.tagName === 'BUTTON') && event.target.classList.contains('popup__close')) {
```

Comment:
don't use deligation here, set on the card when you open it, and remove the listener when you close the card. You know how to do it from this weeks meeting.

Reply:
done